service/src/main/java/org/cbioportal/service/impl/SampleServiceImpl.java

```diff
@@ -190,4 +196,30 @@ private void processSamples(List<Sample> samples, String projection) {
             });
         }
     }
+
+    private void processSamplesSpark(List<Sample> samples, String projection) {
```

Comment:
Is the only difference here wit processSamples() the call to the spark repos?  If so, suggest passing a ref to the repos into a generalized method.

Reply:
That is not the only difference. It's the persistence interface returning internalIds which is something we cannot get from Parquet.

Reply:
See comment above.